execution/scan/matrix_selector.go

```diff
@@ -288,3 +309,112 @@ loop:
 
 	return out, nil
 }
+
+// matrixIterSlice populates a matrix vector covering the requested range for a
+// single time series, with points retrieved from an iterator.
+//
+// As an optimization, the matrix vector may already contain points of the same
+// time series from the evaluation of an earlier step (with lower mint and maxt
+// values). Any such points falling before mint are discarded; points that fall
+// into the [mint, maxt] range are retained; only points with later timestamps
+// are populated from the iterator.
+// TODO(fpetkovski): Add max samples limit.
+func selectExtPoints(it *storage.BufferedSeriesIterator, mint, maxt int64, out []promql.Point, functionName string, extLookbackDelta int64) ([]promql.Point, error) {
+	extRange := function.IsExtFunction(functionName)
```

Comment:
Can we assume we always want the extended range? The function is called `selectExtPoints` already.

Reply:
Removed `ext` relevant logic in the function here: https://github.com/thanos-community/promql-engine/pull/216/commits/63d82031eff1b14f8312b0613ca8f650826aacba